internal/service/alarms/internal/db/repo/alarms_repository.go

```diff
@@ -357,3 +357,78 @@ func getGetAlertFingerPrintAndStartAt(am *api.AlertmanagerNotification) []bob.Ex
+	m := models.AlarmEventRecord{}
+	dbTags := utils.GetAllDBTagsFromStruct(m)
+	queryMods := []bob.Mod[*dialect.SelectQuery]{
+		sm.Columns(utils.GetColumnsAsAny(utils.GetColumns(m, []string{
+			"AlarmEventRecordID", "AlarmDefinitionID", "ProbableCauseID",
+			"AlarmRaisedTime", "AlarmChangedTime", "AlarmClearedTime",
+			"AlarmAcknowledgedTime", "AlarmAcknowledged", "PerceivedSeverity",
+			"Extensions", "ObjectID", "ObjectTypeID",
+			"NotificationEventType", "AlarmSequenceNumber",
+		}))...),
+		sm.From(m.TableName()),
+	}
+
+	// Start with the base condition
+	// Collect all events that haven't been sent to the subscriber yet
+	whereClause := psql.Quote(dbTags["AlarmSequenceNumber"]).GT(psql.Arg(subscription.EventCursor))
+	// If we have a filter, add it to the WHERE clause to reduce further
+	if subscription.Filter != nil {
+		whereClause = psql.And(
+			whereClause,
+			psql.Quote(dbTags["NotificationEventType"]).NE(psql.Arg(subscription.Filter)),
+		)
+	}
+	// Add WHERE and ORDER BY clauses
+	queryMods = append(queryMods,
+		sm.Where(whereClause),
+		sm.OrderBy(dbTags["AlarmSequenceNumber"]).Asc(),
+	)
+
+	// Build final query
+	query := psql.Select(queryMods...)
+
+	sql, params, err := query.Build()
+	if err != nil {
+		return []models.AlarmEventRecord{}, fmt.Errorf("failed to build GetAlarmsForSubscription query: %w", err)
+	}
+
+	records, err := utils.ExecuteCollectRows[models.AlarmEventRecord](ctx, ar.Db, sql, params)
+	if err != nil {
+		return []models.AlarmEventRecord{}, fmt.Errorf("failed to execute GetAlarmsForSubscription query: %w", err)
+	}
+
+	if len(records) > 0 {
+		slog.Info("Successfully got alarms for subscription", "alarm count", len(records), "Subscription", subscription.SubscriptionID)
+	}
+	return records, nil
+}
+
+// UpdateSubscriptionEventCursor update a given subscription event cursor with a alarm sequence value
+func (ar *AlarmsRepository) UpdateSubscriptionEventCursor(ctx context.Context, subscription models.AlarmSubscription, latestSequence int64) error {
+	m := models.AlarmSubscription{}
+	dbTags := utils.GetAllDBTagsFromStruct(m)
+	query := psql.Update(
+		um.Table(m.TableName()),
+		um.SetCol(dbTags["EventCursor"]).ToArg(latestSequence),
+		um.Where(psql.Quote(dbTags["SubscriptionID"]).EQ(psql.Arg(subscription.SubscriptionID))),
+		um.Returning(dbTags["EventCursor"]),
+	)
+
+	sql, params, err := query.Build()
+	if err != nil {
+		return fmt.Errorf("failed to build UpdateSubscriptionEventCursor query: %w", err)
+	}
+
+	record, err := utils.ExecuteCollectExactlyOneRow[models.AlarmSubscription](ctx, ar.Db, sql, params)
+	if err != nil {
+		return fmt.Errorf("failed to execute UpdateSubscriptionEventCursor query: %w", err)
+	}
+
+	slog.Info("Successfully updated subscription with new event cursor", "subscription", subscription.SubscriptionID, "from", subscription.EventCursor, "to", record.EventCursor)
+	return nil
+}
```

Comment:
I think this could be replaced with a call to `utils.Update` similar to this:

```
return utils.Update(ctx, ar.Db, subscription.SubscriptionID, subscription, []string{'EventCursor"}...)
```